### PR TITLE
changes confusion to be less RNG garbage

### PR DESCRIPTION
**Pull request by Delimusca**
this makes confusion cause a mob to walk generally in the direction they're trying to go, based on a probability determined by how confused they are.
likely to affect the function of confusion items like flashes, because i balanced it around getting drunk and forgot about those.
not sure if i should center the probs around 10-30 and nerf ethanol, or increase the effects of everything else.

### CODE REVIEW

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -200,8 +200,15 @@
 							M.animate_movement = 2
 							return
 
-		if(mob.confused && IsEven(world.time))
-			step(mob, pick(cardinal))
+		if(mob.confused)
+			if(mob.confused > 100)
+				step(mob, pick(cardinal))
+			else if(prob(mob.confused / 2))
```
Comment (RemieRichards) [inline code/modules/mob/mob_movement.dm]: nitpick, replace ```/ 2``` with ```* 0.5``` mathematically the same, but faster for code to execute.

**Comment by Anonus** [comment]
* MARGINALLY LESS RNG GARBAGE

File: code/modules/mob/mob_movement.dm (hunk 1 above)
Comment (Delimusca) [inline code/modules/mob/mob_movement.dm]: good point, i actually used bitshifts at first for that reason, honk.

Comment (RemieRichards) [inline code/modules/mob/mob_movement.dm]: honk honk, code speeeeeeeeeed.

**Comment by Iamgoofball** [comment]
>removing the point of confusion

are you fucking kidding me

**Comment by Fox-McCloud** [comment]
:+1: 

Goon did this a while back; if your confusion level is super high, it's more or less the same as it currently is; if it's low, it happens less often and you're far more likely to go int he direction you intend.

**Comment by ike709** [comment]
thank you
:+1:

**Comment by Delimusca** [comment]
changed the numbers a bit, tried to center it around 30. still feels like lower confusion levels should have a greater effect, but i couldn't think of a clean way to do that.

**Pull request merged**